delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/model/ArchiveReference.scala

```diff
@@ -74,4 +79,57 @@ object ArchiveReference {
       ref.original.toString + project.toString + path.toString + representation.toString
   }
 
+  implicit final val archiveReferenceJsonLdDecoder: JsonLdDecoder[ArchiveReference] = {
+    // custom decoder for resource ref to rev and tag references from separate fields
+    implicit val resourceRefDecoder: JsonLdDecoder[ResourceRef] =
+      (cursor: ExpandedJsonLdCursor) =>
+        for {
+          iri    <- cursor.downField(nxv + "resourceId").get[Iri]
+          revOpt <- cursor.downField(nxv + "rev").get[Option[Long]]
+          tagOpt <- cursor.downField(nxv + "tag").get[Option[TagLabel]]
+          value  <- (revOpt, tagOpt) match {
+                      case (None, None)      => Right(ResourceRef.Latest(iri))
+                      case (Some(rev), None) => Right(ResourceRef.Revision(iri, rev))
+                      case (None, Some(tag)) => Right(ResourceRef.Tag(iri, iri, tag))
+                      case _                 =>
+                        Left(ParsingFailure("An archive resource reference cannot use both 'rev' and 'tag' fields."))
```

Comment:
A `ResourceRef` cannot use both 'rev' and 'tag' fields

Reply:
Again, the message is specific to archives because of the custom encoding. A user of archive would have no knowledge of `ResourceRef`.